src/index.js

```diff
@@ -0,0 +1,235 @@
+  "Warm",
+  [70, 79],
+  "orange",
+  "🌸🌿🌼🌷🌻🌿_☘️🌱_🌻🌷"
+);
+const mild = new Temperature("Mild", [60, 69], "gray", "🗻🗻🗻🗻🗻🗻🗻🗻🗻🗻🗻🗻");
+const chilly = new Temperature(
+  "Chilly",
+  [50, 59],
+  "blue",
+  "🌾🌾_🍃_🪨__🛤_🌾🌾_🍃"
+);
+const cold = new Temperature(
+  "Cold",
+  [0, 49],
+  "lightblue",
+  "🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲"
+);
+
+const temperatureCollection = {
+  hot: hot,
+  warm: warm,
+  mild: mild,
+  chilly: chilly,
+  cold: cold,
+};
+
+changeTempDisplay(cold);
+
+const fullSunny = new Weather(
+  "Full Sunny",
+  "🌞🌞🌞🌞🌞🌞🌞🌞🌞🌞🌞🌞",
+  "red"
+);
+const sunny = new Weather(
+  "Sunny",
+  "☁️ ☁️ ☀️☀️☀️☀️☀️☀️☀️ ☁️ ☁️",
+  "rgb(221, 255, 255)"
+);
+const cloudy = new Weather(
+  "Cloudy",
+  "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️",
+  "lightgrey"
+);
+const rainy = new Weather(
+  "Rainy",
+  "🌧🌈⛈🌧💧⛈🌧🌦🌧💧🌧🌧",
+  "lightblue"
+);
+const snowy = new Weather(
+  "Snowy",
+  "🌨❄️🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨",
+  "lightsteelblue"
+);
+const apocalyptic = new Weather(
+  "Apocalyptic",
+  "🧟‍♀️🧟‍♂️🧟‍♀️🌫🧟‍♂️🌅🧟‍♂️🌫🧟‍♀️🧟‍♂️🧟‍♀️",
+  "rgb(11, 247, 46)"
+);
+
+const skyCollection = {
+  apocalyptic: apocalyptic,
+  snowy: snowy,
+  rainy: rainy,
+  cloudy: cloudy,
+  sunny: sunny,
+  fullSunny: fullSunny,
+};
+
+const landscapeCollection = {
+  apocalyptic: apocalyptic,
+  snowy: snowy,
+  rainy: rainy,
+  cloudy: cloudy,
+  sunny: sunny,
+  fullSunny: fullSunny,
+};
+
+let tempNumber = 0;
+const temperature = document.getElementById("tempValue");
+temperature.textContent = `${tempNumber}° F`;
+
+const increaseTempButton = document.getElementById("increaseTempControl");
+increaseTempButton.addEventListener("click", () => {
+  tempNumber++;
+  temperature.textContent = `${tempNumber}° F`;
+  updateWeatherBasedOnTemperature();
+});
+
+const decreaseTempButton = document.getElementById("decreaseTempControl");
+decreaseTempButton.addEventListener("click", () => {
+  tempNumber--;
+  temperature.textContent = `${tempNumber}° F`;
+  updateWeatherBasedOnTemperature();
+});
+
+const cityNameInput = document.getElementById("cityNameInput");
+const headerCityName = document.getElementById("headerCityName");
+
+cityNameInput.addEventListener("input", () => {
+  headerCityName.textContent = cityNameInput.value;
+});
+
+const skySelect = document.getElementById("skySelect");
+skySelect.addEventListener("input", (event) => {
+  const selectedSky = skySelect.value;
+  changeSky(skyCollection[selectedSky]);
+});
+
+const updateWeatherBasedOnTemperature = async () => {
+  for (const temperatureKey in temperatureCollection) {
+    const temperature = temperatureCollection[temperatureKey];
+    if (
+      tempNumber >= temperature.tempRange[0] &&
+      tempNumber <= temperature.tempRange[1]
+    ) {
+      changeTempDisplay(temperature);
+      break;
+    }
+  }
+};
+
+const resetButton = document.getElementById("cityNameReset");
+resetButton.addEventListener("click", () => {
+  cityNameInput.value = "Austin";
+  headerCityName.textContent = "Austin";
+});
+
+const axiosGetWeather = (cityName) => {
```

Comment:
This function seems incomplete? Where it is called, it's called with an `await` keyword, which sets up the expectation that it will return a promise, which this function doesn't. It also appears to be named in such a way where it will get weather data but looks like it attempts to get both location and weather data. I'd recommend modifying it to just do what it's named for and change it to an async function so that the usage makes more sense. E.g.
```js
const axiosGetWeather = async ({ latitude, longitude }) => {
  const response = await axios.get("http://localhost:5000/weather", {
    params: {
      lat: latitude,
      lon: longitude,
    },
  });

  return response.data.main.temp;
};
```
This approach assumes that there will be a function that returns an object with `lattitude` and `longitude` coordinates. Following the pattern you laid out, you could define the function you called (that currently doesn't exist), like:
```js
const axiosGetCoordinates = async (cityName) => {
  const response = await axios.get("http://127.0.0.1:5000/location", {
    params: {
      q: cityName,
    },
  });

  const { lat: latitude, lon: longitude } = response.data[0];

  return { latitude, longitude };
};
```
Then the usage would be something like how you have it on lines 209-210:
```js
const coordinates = await axiosGetCoordinates(cityNameInput.value);
const temperatureValue = await axiosGetWeather(coordinates);
```